File: Data Structures & Algorithms/meeting-schedule/submission-0.py
"""
Definition of Interval:
class Interval(object):
    def __init__(self, start, end):
        self.start = start
        self.end = end
"""

class Solution:
    def canAttendMeetings(self, intervals: List[Interval]) -> bool:
        intervals.sort(key=lambda x: x.start)
        mx = float("-inf")
        for meeting in intervals :
            start ,end = meeting.start , meeting.end
            if start >= mx:
                mx = end
            else:
                return False

        return True

File: Data Structures & Algorithms/meeting-schedule-ii/submission-0.js
class Solution {
  /**
   * @param {Interval[]} intervals
   * @returns {number}
   */
  minMeetingRooms(intervals) {
    const startArray = intervals.sort((a, b) => a.start - b.start).map((el) => el.start);
    const endArray = intervals.sort((a, b) => a.end - b.end).map((el) => el.end);
    let startPtr = 0, endPtr = 0, counter = 0, ans = 0;
    while (startPtr < intervals.length && endPtr < intervals.length) {
      const curStart = startArray[startPtr], curEnd = endArray[endPtr];
      if (curStart >= curEnd) {
        counter--;
        endPtr++;
      } else {
        counter++;
      }
      startPtr++;
      ans = Math.max(ans, counter);
    }
    return ans;

  }
}
